Unknown feature names are not reported if there are other errors

I just added an impl that I thought should be valid now with https://github.com/rust-lang/rfcs/pull/2451, but I got a compiler error. So I found the RFC and copied the feature name `re_rebalancing_coherence` from it. However, I still got the compile error, so I spent a while trying to figure out if the impl was really valid. Eventually, I tried commenting out the impl, and then (thanks to #52644), the compiler told me the feature was unknown (it has been renamed to `re_rebalance_coherence`). The impl was valid using the correct feature name. I didn't get the error saying the feature name was invalid because of the error caused by the feature name being invalid.

I think the compiler should report unknown feature names even if there are other errors. For example: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=e46fde7bf4076220df3005d702df9cbb doesn't report the feature name is invalid. With box syntax, it suggests "help: add #![feature(box_syntax)] to the crate attributes to enable" but the coherence error didn't.

Feel free to close this issue if you don't think it is worth fixing.

Current output:
```
error: `box_syntax` has been removed
 --> src/main.rs:5:13
  |
5 |     let x = box 5;
  |             ^^^^^
  |
help: use `Box::new()` instead
  |
5 -     let x = box 5;
5 +     let x = Box::new(5);
  |

error[E0635]: unknown feature `box_syntax2`
 --> src/main.rs:2:12
  |
2 | #![feature(box_syntax2)]
  |            ^^^^^^^^^^^
```
Tried [other](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=c05b6e53ec04d2dd1f3e60479c16ce9d) [cases](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=9da687f952dc8950a18ac90914f143b0) to check that the error is still being emitted even if there are other errors.

--- a/tests/ui/feature-gates/unknown-feature-with-other-errors-58390.rs
+++ b/tests/ui/feature-gates/unknown-feature-with-other-errors-58390.rs
@@ -1,0 +1,16 @@
+//! Regression test for <https://github.com/rust-lang/rust/issues/58390>.
+//!
+//! An unknown `#![feature(..)]` name used to be silently ignored whenever the crate had any
+//! other error, because the check only ran during stability checking. Both errors must be
+//! reported.
+
+#![feature(this_feature_does_not_exist)] //~ ERROR unknown feature `this_feature_does_not_exist`
+
+struct Foo;
+
+trait Bar {}
+
+impl Bar for Foo {}
+impl Bar for Foo {} //~ ERROR conflicting implementations of trait `Bar` for type `Foo`
+
+fn main() {}

--- a/tests/ui/feature-gates/unknown-feature-with-other-errors-58390.stderr
+++ b/tests/ui/feature-gates/unknown-feature-with-other-errors-58390.stderr
@@ -1,0 +1,18 @@
+error[E0635]: unknown feature `this_feature_does_not_exist`
+  --> $DIR/unknown-feature-with-other-errors-58390.rs:7:12
+   |
+LL | #![feature(this_feature_does_not_exist)]
+   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error[E0119]: conflicting implementations of trait `Bar` for type `Foo`
+  --> $DIR/unknown-feature-with-other-errors-58390.rs:14:1
+   |
+LL | impl Bar for Foo {}
+   | ---------------- first implementation here
+LL | impl Bar for Foo {}
+   | ^^^^^^^^^^^^^^^^ conflicting implementation for `Foo`
+
+error: aborting due to 2 previous errors
+
+Some errors have detailed explanations: E0119, E0635.
+For more information about an error, try `rustc --explain E0119`.

